decisions/remote-server-backup.md

```diff
@@ -0,0 +1,57 @@
+# Remote Server Backup
+
+- *Date*: 2022-12-7
+- *Deciders*: 
+- *Status*: 
+- *Outcome*: 
+- *Related Material*: [Backup Issue](https://github.com/openmsupply/open-msupply/issues/767)
+
+## Context
+
+It's important to back up remote server data so that it can be restored in case of corruption/loss. This KDD is to discuss remote synchronised server backup.
+
+In existing 4d mSupply, for remote servers, we have backup files and current journal file. There is no way to re-initialise remote desktop mSupply sites, thus we must rely on local/dropbox backups. There have been cases with quite catastrophic outcomes where backup was restored incorrectly, causing major data synchronisation issues.
+
+### Wrongly Restored
+* Wrong backup (not the latest)
+* Forget to integrate journal
+* Journal data corruption
+
+### Major data syncrhonisation issues
+
+When remote data is not `in sync` with central data (i.e. older backup was used), we almost certainly have a ledge discrepancy that can persist and is hard to fix correctly. 
+
+**Example:**
+
+Consider that remote and central server data is synchronised, and we have the following ledger for stockline: stockline quantity = 10, transaction1 = +15, transaction2 = -2 transaction3 = -3. 15 - 2 - 3 = 10 thus ledge is correct. Now if remote server backup was restored not including transaction 3, stockline quantity would be 13, and any further transaction or create ledger discrepancy for this stockline on central server.
+
+## Options
+
+### Option 1 - Local backup
+
+Local backup with journal
+
+*Pros:*
+- Can be restored easier with low bandwidth environment
+- Reduce data loss if done correctly in cases where internet was not available and sync not possible
+
+*Cons:*
+- Hard to set up correctly and to maintain (for both sqlite and postgres)
```

Comment:
I don't understand the difficulty - so I'm probably missing something. For either db, shutting down the server and doing a file level copy would back it up. For pg, can we not run `pg_dump` from a scheduler?

Also don't agree that a database backup necessarily requires large disk space. Depends on the size of the db and the rotation strategy

Reply:
I think my point about 'hard to set up and maintain' was to do with backup up and restoring `wal` rather then file snapshot.

> Also don't agree that a database backup necessarily requires large disk space. Depends on the size of the db and the rotation strategy

Oki, i'l change that to be `Larger` and Option 3 would be almost purely `wal` which would take a lot less space

Reply:
I've changed that it's `harder`, but I generally think it's hard because it creates a lot of use cases where things can go wrong:
* Are we restoring correct data or data from old backup
* We have to wait for shut down of server before doing file level copy (how do we setup reliable scheduler for this in Windows). What if server doesn't shut down ? Would snapshot file backup be in undefined state ?
* How frequent should our backup be, if it's once a day would we be missing whole day of work (what if it was a stocktake day)
* If we use wal what if we forget to integrate wal ?